### PR TITLE
#184581171 A Seller Should see statistics in a given time-frame

### PR DESCRIPTION
**Pull request by AbdulKhaliq59**
#### What does this PR do?
This PR give a seller sales statistics in a given time frame

#### Description of Task to be completed?
- [x] Create endpoint for getting statistics  and filter it for his/her sales
- [x] Create documentation for all such endpoints
- [x] Test such endpoints

#### How should this be manually tested?
- Clone this repo git clone [e-comm-team-amigos-bn](https://github.com/atlp-rwanda/e-comm-team-amigos-bn)
- Switch your working directory cd `e-comm-team-amigos-bn`
- Create an .env file and update it with `.env-example`
- Install all the dependencies by running `npm install`
- Then run the project by typing `npm start`

#### Endpoints to be tested:
-  GET `/product/sales/stats`

#### Any background context you want to provide?
- Have Node Js installed in your local machine
- ...........

#### What are the relevant pivotal tracker/Trello stories?

Story ID: [#184581171](https://www.pivotaltracker.com/story/show/184581171)

#### Screenshots (if appropriate)
screencapture-localhost-4000-docs-2023-04-05-13_02_13
Screenshot 2023-04-05 at 13 05 09

### CODE REVIEW

File: src/controllers/authentication.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const createUser = async (req, res) => {
             token,
         });
     } catch (error) {
-        console.log(error)
+        console.log(error);
```
Comment (nshutijonathan) [inline src/controllers/authentication.js:59]: remove console.log and return instead

File: src/controllers/statistics.controller.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        );
+
+        return res.json({
+            totalRevenue,
+            numOrders: Object.keys(numOrders).length,
+            productsSold,
+            topSellingProducts,
+        });
+    } catch (error) {
+        console.log(error);
```
Comment (nshutijonathan) [inline src/controllers/statistics.controller.js]: return error.message

**Review by nshutijonathan — COMMENTED** [review]
LGTM

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Review by ericndungutse — APPROVED** [review]

**Review by Mystica52 — APPROVED** [review]

**Review by Rwibutso — APPROVED** [review]

**Review by GasanaJr — APPROVED** [review]
LGTM

**Pull request merged**